include/storage/stream_flash.h

```diff
@@ -77,12 +80,13 @@ struct stream_flash_ctx {
  *             If this is '0', the size will be set to the total size
  *             of the flash device minus the offset.
  * @param cb Callback to be invoked on completed flash write operations.
- *
+ * @param id Unique identifier for this stream, used to store write progress.
+ *           Can be set to NULL to disable progress saving for the stream.
  * @return non-negative on success, negative errno code on fail
  */
 int stream_flash_init(struct stream_flash_ctx *ctx, const struct device *fdev,
```

Comment:
Instead of changing the signature, we should extend the API, to be backwards compatible.